code/modules/mob/living/silicon/ai/ai_programs.dm

```diff
@@ -0,0 +1,1017 @@
+	..()
+	SSeconomy.pack_price_modifier = original_price_mod * (0.95 - (0.05 * upgrade_level))
+
+/datum/ai_program/multimarket_analyser/uninstall(mob/living/silicon/ai/user)
+	..()
+	SSeconomy.pack_price_modifier = original_price_mod
+
+// Light Synthesizer - Fixes lights
+/datum/ai_program/light_repair
+	program_name = "Light Synthesizer"
+	program_id = "light_repair"
+	description = "Replace damaged or missing lightbulbs."
+	nanite_cost = 5
+	power_type = /datum/spell/ai_spell/ranged/light_repair
+	unlock_text = "Light replacer configuration installed."
+
+/datum/spell/ai_spell/ranged/light_repair
+	name = "Light Synthesizer"
+	desc = "Replace damaged or missing lightbulbs."
+	action_icon = 'icons/obj/janitor.dmi'
+	action_icon_state = "lightreplacer0"
+	ranged_mousepointer = 'icons/mecha/mecha_mouse.dmi'
+	auto_use_uses = FALSE
+	base_cooldown = 30 SECONDS
+	cooldown_min = 5 SECONDS
+	level_max = 5
+	selection_activated_message = "<span class='notice'>You prepare to synthesize a lightbulb...</span>"
+	selection_deactivated_message = "<span class='notice'>You cancel the request.</span>"
+
+/datum/spell/ai_spell/ranged/light_repair/cast(list/targets, mob/user)
+	var/obj/machinery/light/target = targets[1]
+	if(!check_camera_vision(user, target))
+		revert_cast()
+		return
+	if(!istype(target))
+		to_chat(user, "<span class='warning'>You can only repair lights!</span>")
+		revert_cast()
+		return
+	var/mob/living/silicon/ai/AI = user
+	// Handle repairs here since we're using a spell and not a tool
+	target.status = LIGHT_OK
+	target.switchcount = 0
+	target.emagged = FALSE
+	target.on = target.has_power()
+	target.update(TRUE, TRUE, FALSE)
+	AI.program_picker.nanites -= 5
+	AI.play_sound_remote(target, 'sound/machines/ding.ogg', 50)
+	camera_beam(target, "rped_upgrade", 'icons/effects/effects.dmi', 5)
+
+/datum/spell/ai_spell/ranged/rgb_lighting/on_purchase_upgrade()
+	cooldown_handler.recharge_duration = base_cooldown - (spell_level * 5)
+
+// Nanosurgeon Deployment - Uses large numbers of nanites to heal things
+/datum/ai_program/nanosurgeon_deployment
+	program_name = "Nanosurgeon Deployment"
+	program_id = "nanosurgeon_deployment"
+	description = "Heal a crew member with large numbers of robotic nanomachines!"
+	cost = 3
+	nanite_cost = 75
+	power_type = /datum/spell/ai_spell/ranged/nanosurgeon_deployment
+	unlock_text = "Surgical nanomachine firmware installation complete!"
+
+/datum/spell/ai_spell/ranged/nanosurgeon_deployment
+	name = "Nanosurgeon Deployment"
+	desc = "Heal a crew member with large numbers of robotic nanomachines!"
+	action_icon = 'icons/obj/surgery.dmi'
+	action_icon_state = "scalpel_laser1_on"
+	ranged_mousepointer = 'icons/mecha/mecha_mouse.dmi'
+	auto_use_uses = FALSE
+	base_cooldown = 300 SECONDS
+	cooldown_min = 30 SECONDS
+	level_max = 8
+	selection_activated_message = "<span class='notice'>You prepare to order your nanomachines to perform surgery...</span>"
```

Comment:
I would avoid using the word surgery here as that has a very specific meaning, and it may lead people to think they need to actually perform surgery steps as part of the ability